Scripts/Source/minai_Sex.psc

```diff
@@ -663,96 +663,110 @@ EndFunction
 
 
 
-Event OnOstimOrgasm(string eventName, string strArg, float numArg, Form sender)
-  actor akActor = sender as actor
-EndEvent
-
-
-
 Event OStimManager(string eventName, string strArg, float numArg, Form sender)
+  string sceneId = strArg
   int ostimTid = numArg as int
-  Main.Info("oStim eventName: "+eventName+", strArg: "+strArg);
-  if (eventName=="ostim_thread_start")
-    string sceneName=OThread.GetScene(ostimTid);
-    bool isRunning=OThread.IsRunning(ostimTid);
-    Actor[] actors =  OThread.GetActors(ostimTid);
-    string actorString;
+  Main.Debug("oStim eventName: "+eventName+", strArg: "+strArg+", numArg: "+numArg as int+" sender: "+sender as actor )
+  if (eventName == "ostim_thread_start")
+    bool isRunning = OThread.IsRunning(ostimTid)
+    Actor[] actors = OThread.GetActors(ostimTid)
+    string actorString
     int i = actors.Length
     bool playerInvolved=false
-    while(i > 0)
+    if isRunning
+      while(i > 0)
```

Comment:
This can be replaced with a check to actors.Find(playerRef) >= 0

Reply:
ok, is that really more efficient?